### PR TITLE
Add Rest endpoint to check if email exists

### LINKED ISSUE #925: Create endpoint to check if email exists
**Is your feature request related to a problem? If so, please describe.**
When we log in on the frontend, we ask for the email and password together. If we ask for them separately, we can provide a nicer user experience by letting the user know which (out of email and password) field they typed in was incorrect

**Describe the solution you'd like**
Create an endpoint in the [UserController](https://github.com/Project-Books/book-project/blob/main/backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java) that uses our [service method](https://github.com/Project-Books/book-project/blob/main/backend/book-app/src/main/java/com/karankumar/bookproject/account/service/UserService.java#L138) to see whether the email exists.

If the email does not exist, we should return a 404. If the email does exist, we can return 200 OK. This should be a GET method.

**Additional context**
- Be sure to read the contributing guidelines first!
- Discuss your implementation approach with us before we assign you to it
- Provide us with an estimate on how long you think it will take you to complete
- Branch from `main` and set `main` as the destination branch in your pull request

If you need help with anything, we'll be happy to help you over a [GitHub Q&A discussion](https://github.com/Project-Books/book-project/discussions/categories/q-a) or our [Slack workspace](https://teambookproject.slack.com/join/shared_invite/zt-punc8os7-Iz9PTCAkYcO_0S~XwtO5_A#/shared-invite/email)

### PR DESCRIPTION
**Pull request by kuruvasatya**
## Summary of change

Added new rest endpoint in the UserController.java class to check if email exists or not

## Related issue

Closes #925

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

### CODE REVIEW

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for adding this endpoint! Could you also please write some tests for this (check that we return 200 OK on success and 404 on failure)?

**Comment by kuruvasatya** [comment]
Hi @knjk04, for some reasons MockMvc is not working, it is blocking me from writing test, could you help regarding the same?

**Comment by knjk04** [comment]
Hi @kuruvasatya, sure. I can help you with this. How come you closed this pull request?

**Comment by kuruvasatya** [comment]
Hey @knjk04, that was by mistake, i am able to test the service, but for testing controller I think we need MockMvc, that seems to be not working, please help me

**Comment by kuruvasatya** [comment]
Hey @knjk04 any Updates, its been days

File: backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java [hunk 1]
```diff
@@ -88,6 +90,16 @@ public User getUser(@PathVariable Long id) {
         );
   }
 
+  @GetMapping("/email/{email}")
+  public ResponseEntity<String> checkIfEmailExists(@PathVariable String email) {
+      Optional<User> user = userService.findUserByEmail(email);
```
Comment (mslowiak) [inline backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java:95]:
```
userService.findUserByEmail(email)
    .map(user-> ResponseEntity.ok())
    .orElse(ResponseEntity.status(HttpStatus.NOT_FOUND).body(EMAIL_NOT_FOUND));
 ```

**Review by mslowiak — CHANGES_REQUESTED** [review]

**Comment by knjk04** [comment]
Hi @kuruvasatya, so sorry for the delay & thanks for chasing up. I didn't get a chance to implement this, but I'll do it now.

File: backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java (hunk 1 above)
Comment (knjk04) [reply backend/book-app/src/main/java/com/karankumar/bookproject/account/controller/UserController.java:95]: Thanks, @mslowiak! That's a lot more concise!

**Review by knjk04 — APPROVED** [review]
Thanks!

**Pull request merged**